### PR TITLE
Initialize Local Page Cache

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/6827

Problem Summary:

### What is changed and how it works?

Init local page cache according to the configured cache dir.

Note: Due to our implementation, it is not possible to disable the local page cache. When `storage.remote.cache` is not configured, data directory will be used as the cache directory and there will be no capacity limit.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/Context.cpp [hunk 1]
```diff
@@ -166,8 +167,12 @@ struct ContextShared
     PageStorageRunMode storage_run_mode = PageStorageRunMode::ONLY_V3;
     DM::GlobalStoragePoolPtr global_storage_pool;
 
-    /// The PS instance available on Write Node.
+    /// The PS instance available on Write Node, or we are not in disaggregated mode.
     UniversalPageStorageServicePtr ps_write;
+    /// The PS instance available on Disaggregated Read Node.
+    UniversalPageStorageServicePtr ps_rn_page_cache;
+    /// The page cache in disaggregated compute node, used to cache pages.
+    DB::DM::Remote::RNLocalPageCachePtr rn_page_cache;
```
Comment (flowbehappy) [inline dbms/src/Interpreters/Context.cpp:175]:
```suggestion
    /// The following members are only available in read-write disaggregated mode
    ///
    /// The PS instance available on Write Node.
    UniversalPageStorageServicePtr ps_write;
    /// The PS instance available on Read Node.
    UniversalPageStorageServicePtr ps_rn_page_cache;
    /// The page cache in Read Node. It uses ps_rn_page_cache as storage to cache page data to local disk based on the LRU mechanism.
    DB::DM::Remote::RNLocalPageCachePtr rn_page_cache;
```

File: dbms/src/Server/Server.cpp [hunk 2]
```diff
@@ -1136,7 +1138,14 @@ int Server::main(const std::vector<std::string> & /*args*/)
     global_context->initializeGlobalStoragePoolIfNeed(global_context->getPathPool());
     LOG_INFO(log, "Global PageStorage run mode is {}", static_cast<UInt8>(global_context->getPageStorageRunMode()));
 
-    global_context->initializeWriteNodePageStorageIfNeed(global_context->getPathPool());
+    if (global_context->isDisaggregatedStorageMode())
+        global_context->initializeWriteNodePageStorageIfNeed(global_context->getPathPool());
```
Comment (JaySon-Huang) [inline dbms/src/Server/Server.cpp:1142]: 1. Now if user want to use S3, they won't set `flash.disaggregated_mode`
2. `isDisaggregatedStorageMode` can not distinguish "non-disagg tiflash instance" or "disagg-write-node tiflash instance" either
https://github.com/pingcap/tiflash/blob/8794db4ebc003d8a12e02bf0d2aa889f84ab7a19/dbms/src/Interpreters/Context.h#L515-L519

**Review by flowbehappy — APPROVED** [review]
LGTM

File: dbms/src/Server/Server.cpp (hunk 2 above)
Comment (JaySon-Huang) [reply dbms/src/Server/Server.cpp:1142]: `isDisaggregatedStorageMode` can not distinguish "non-disagg tiflash instance" or "disagg-write-node tiflash instance" is by design

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**